src/App.scss

```diff
@@ -73,84 +104,109 @@ body {
     margin-left: 50px;
 }
 
-.App__program-section {
-    margin-top: 1.5rem;
+.App__world-selector-container {
+    width: 6rem;
+    grid-column-start: 4;
+    grid-column-end: 5;
+    grid-row-start: 2;
+    grid-row-end: 3;
 }
 
 .App__command-palette {
     background-color: #F1F1F1;
-    height: 100%;
-    padding-top: 2rem;
     text-align: center;
+    margin: 0 0.5rem;
+    grid-column-start: 2;
+    grid-column-end: 3;
+    grid-row-start: 2;
+    grid-row-end: 4;
+    overflow: hidden;
 }
 
 .App__command-palette-heading {
     font-weight: bold;
     font-size: inherit;
-    margin: 0 0 2rem 0;
+    margin: 2rem 0;
+}
+
+.App__command-palette-command-container {
+    width: 100%;
+    height: 100%;
+    padding: 1rem;
+    overflow-y: scroll;
 }
 
 .App__command-palette-commands {
-    // width = #items * itemWidth + gapWidth * (#items - 1)
-    width: 14rem;
-    margin: 3rem auto 0 auto;
+    width: 6rem;
     display: grid;
-    grid-template-columns: repeat(3, 4rem);
+    grid-template-columns: repeat(2, 2.5rem);
     grid-gap: 1rem;
 }
 
-.App__playAndShare-container {
+.App__program-block-editor {
```

Comment:
I see you have updated grid-column-end to include the width of the world selector component you have added, but we don't need to update grid-row-end for the program block editor, because there was no changes to the rows of overall layout. With this new grid-row-end value, ProgramBlockEditor and PlayAndShare container are fighting for the row 4 -- let's change value of 'grid-row-end' back to what it was before.

`
 grid-row-end: 4;
`

Reply:
Right.  Fixed.